src/httpProblemDetail.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpProblemDetailDefinition } from './httpProblemDetailDefinition';
+import { URI, HttpStatusCode } from './types';
+
+const forbiddenDetailNames = ['type', 'status', 'title'];
+
+const defaultType = 'about:blank';
+
+export class HttpProblemDetail {
+  public readonly type: URI;
+  public readonly title: string;
+  public readonly status: HttpStatusCode;
+  private additionalDetails: Record<string, any>;
```

Comment:
I guess it should be object which is easily serializable to JSON so instead of `Record` I would add more specific type:
```
type Value = string | number | boolean
type ObjectOrArrayOrValue = { [key: string]: ObjectOrArrayOrValue } | Array<ObjectOrArrayOrValue> | Value
type JsonSerializableObject = { [key: string]: ObjectOrArrayOrValue }
export default JsonSerializableObject
```
like it was in Revote here:
https://github.com/DroidsOnRoids/ReVote-API/blob/master/src/EventsBus/types.ts#L9

Reply:
Btw this also should be `readonly`.